### ISSUE TITLE
Today's PV generation reading in HA is different to what SEMS reports

### ISSUE THREAD

**Issue by rinners** [home-assistant/home-assistant.github.io#22637]
### Feedback

Good day,

I've recently installed the Goodwe HA integration and am using it with a GW5048D-ES and the "Today's PV generation" values seem to be higher than what SEMS is reporting.

I created helpers for doing the integration sum and so far it seems that the Left Riemann sum is the closest match to the SEMS value.

Would appreciate if this could be looked and if I'm doing something wrong please let me know.

Thanks,
Darren

### URL

https://www.home-assistant.io/integrations/goodwe/

### Version

2022.5.0

### Additional information

![WhatsApp Image 2022-05-05 at 9 29 50 AM](https://user-images.githubusercontent.com/48795308/166880925-7915a5d4-47a2-4761-918a-63ec3b55eeb8.jpeg)
![2022-05-05 09_28_42-Window](https://user-images.githubusercontent.com/48795308/166880449-cbd2e3c2-6eea-4172-b2b3-f1b72c96b748.png)
![2022-05-05 09_28_02-Window](https://user-images.githubusercontent.com/48795308/166880465-4ead3584-a62c-4c68-96ab-0ce120483c69.png)
![2022-05-05 09_25_34-Window](https://user-images.githubusercontent.com/48795308/166880468-d13d78b0-3233-4a8c-b775-299917bef389.png)

**Comment by frenck** [comment]
Hi there @rinners 👋 

It seems like you want to report an issue with an Home Assistant integration, however, you have submitted documentation feedback instead.

Please report issues with integrations here:

<https://github.com/home-assistant/core/issues>

Thanks! 👍 

../Frenck

**Comment by mletenay** [comment]
'Today's PV Generation' sensor value is provided directly by the inverter.
There is no calculation or conversion involved in HA.
The SEMS portal likely performs its own calculation (i.e. left Riemann) and the sums and graphs are not based on this "total" value but on current power measurements.
Which be definition may not be precise, since those measurements are reported to SEMS with quite low frequency.
In short, the 'Today's PV Generation' is expected to provide the most precise values, SEMS is just approximation.
